packages/rocketchat-ui/client/views/app/room.js

```diff
@@ -965,9 +929,64 @@ Template.room.onCreated(function() {
 	});
 
 	this.sendToBottomIfNecessary = () => {};
+
+	const hideMessagesOfType = [];
+	settings.collection.find({ _id: /Message_HideType_.+/ }).forEach(function(record) {
+		let types;
+		const type = record._id.replace('Message_HideType_', '');
+		switch (type) {
+			case 'mute_unmute':
+				types = ['user-muted', 'user-unmuted'];
+				break;
+			default:
+				types = [type];
+		}
+		return types.forEach(function(type) {
+			const index = hideMessagesOfType.indexOf(type);
+
+			if ((record.value === true) && (index === -1)) {
+				hideMessagesOfType.push(type);
+			} else if (index > -1) {
+				hideMessagesOfType.splice(index, 1);
+			}
+		});
+	});
+
+	const query =
+		{ rid: this.data._id };
+
+	if (hideMessagesOfType.length > 0) {
+		query.t =
+			{ $nin: hideMessagesOfType };
+	}
+
+	const options = {
+		sort: {
+			ts: 1,
+		},
+	};
+
+	this.messages = new ReactiveVar([]);
+
+	const updateMessages = () => {
+		this.messages.set(ChatMessage.find(query, options).fetch());
+	};
+
+	const updateMessagesDebounced = _.debounce(() => {
+		this.messages.set(ChatMessage.find(query, options).fetch());
+	}, 100);
+
+	this.messageObserver = ChatMessage.find(query).observe({
+		added: process.env.TEST_MODE ? updateMessages : updateMessagesDebounced,
```

Comment:
I dont like this approach (using `process.env.TEST_MODE`) but if we gonna use that... I would suggest few changes
```
const updateMessages = () => {
	this.messages.set(ChatMessage.find(query, options).fetch());
};
const updateMessagesDebounced = _.debounce(updateMessages, 100);
```
 more suggestions:
```

const _updateMessages = () => {
	this.messages.set(ChatMessage.find(query, options).fetch());
};
const updateMessages = process.env.TEST_MODE? _.debounce(_updateMessages, 100): _updateMessages ;

this.messageObserver = ChatMessage.find(query).observe({
  added: updateMessages,
  updated: updateMessages,
  removed: updateMessages,
})
```

Reply:
It was removed